src/dag/gc.ts

```diff
@@ -0,0 +1,90 @@
+import type {Hash} from '../hash';
+
+export type HeadChange = {
+  new: Hash | undefined;
+  old: Hash | undefined;
+};
+
+export class GarbargeCollector {
+  private readonly _setRefCount: (hash: Hash, count: number) => Promise<void>;
+  private readonly _getRefCount: (hash: Hash) => Promise<number>;
+  private readonly _deleteChunk: (hash: Hash) => Promise<void>;
+  private readonly _getRefs: (
+    hash: Hash,
+  ) => Promise<readonly Hash[] | undefined>;
+
+  constructor(
+    setRefCount: (hash: Hash, count: number) => Promise<void>,
```

Comment:
this is a lot of function that take a hash, passing them in the right order is a bit finicky, change to accept a single object implementing an interface.

Reply:
I would not do that, objects are heap allocated and thus have some overhead when this is all internal. TS will complain if things are in wrong order and VSCode can show you parameter hints.

Reply:
im thinking an interface that the WriteImpl can just implement.